dags/tower_metrics_dag.py

```diff
@@ -0,0 +1,63 @@
+from datetime import datetime
+
+from airflow.decorators import dag
+
+from dag_content.tower_metrics_content import *
+
+
+@dag(
+    schedule_interval="@weekly",
+    start_date=datetime(2022, 11, 11),
+    catchup=False,
+    default_args={
+        "retries": 3,
+    },
+    tags=["Nextflow Tower Metrics"],
+)
+def tower_metrics_dag():
+    # get needed info from production database for cloining
+    prod_db_info = get_database_info(aws_creds=AWS_CREDS, db_name=DATABASE_NAME)
+    # clone tower database
+    clone_db_info = clone_tower_database(
+        aws_creds=AWS_CREDS,
+        db_name=DATABASE_NAME,
+        clone_name=CLONE_DATABASE_NAME,
+        subnet_group=prod_db_info["subnet_group"],
+        security_group=prod_db_info["security_group"],
+    )
+    # generate new password while clone is spinning up
+    password = generate_random_password(aws_creds=AWS_CREDS)
+    # update cloned database to have new password
+    modify = modify_database_clone(
```

Comment:
It may not be ideal, but it seems like in order to explicitly set dependencies, a task must be instantiated, even if there are no XCOMs to pass along. In this example, `modify_database_clone` does not `return` anything, but creating the object `modify` is necessary to then ensure that modifying the database comes after password generation and before the database is queried.